go/enclave/actors/sequencer.go

```diff
@@ -0,0 +1,307 @@
+package actors
+
+import (
+	"bytes"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"fmt"
+	"time"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	gethlog "github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/obscuronet/go-obscuro/go/common"
+	"github.com/obscuronet/go-obscuro/go/common/log"
+	"github.com/obscuronet/go-obscuro/go/enclave/components"
+	"github.com/obscuronet/go-obscuro/go/enclave/core"
+	"github.com/obscuronet/go-obscuro/go/enclave/crypto"
+	"github.com/obscuronet/go-obscuro/go/enclave/db"
+	"github.com/obscuronet/go-obscuro/go/enclave/mempool"
+)
+
+type sequencer struct {
+	consumer       components.BlockConsumer
+	producer       components.BatchProducer
+	registry       components.BatchRegistry
+	rollupProducer components.RollupProducer
+	rollupConsumer components.RollupConsumer
+
+	logger gethlog.Logger
+
+	hostID            gethcommon.Address
+	chainConfig       *params.ChainConfig
+	enclavePrivateKey *ecdsa.PrivateKey // this is a key known only to the current enclave, and the public key was shared with everyone during attestation
+	mempool           mempool.Manager
+	storage           db.Storage
+	encryption        crypto.TransactionBlobCrypto
+}
+
+func NewSequencer(
+	consumer components.BlockConsumer,
+	producer components.BatchProducer,
+	registry components.BatchRegistry,
+	rollupProducer components.RollupProducer,
+	rollupConsumer components.RollupConsumer,
+
+	logger gethlog.Logger,
+
+	hostID gethcommon.Address,
+	chainConfig *params.ChainConfig,
+	enclavePrivateKey *ecdsa.PrivateKey, // this is a key known only to the current enclave, and the public key was shared with everyone during attestation
+	mempool mempool.Manager,
+	storage db.Storage,
+	encryption crypto.TransactionBlobCrypto,
+) Sequencer {
+	return &sequencer{
+		consumer:          consumer,
+		producer:          producer,
+		registry:          registry,
+		rollupProducer:    rollupProducer,
+		rollupConsumer:    rollupConsumer,
+		logger:            logger,
+		hostID:            hostID,
+		chainConfig:       chainConfig,
+		enclavePrivateKey: enclavePrivateKey,
+		mempool:           mempool,
+		storage:           storage,
+		encryption:        encryption,
+	}
+}
+
+func (s *sequencer) IsReady() bool {
```

Comment:
I think I know why this returns false, but can a comment be added to make it clear ?

Reply:
just needs to be finished/removed